fontforge/svg.c

```diff
@@ -1081,17 +1089,45 @@ int _ExportSVG(FILE *svg,SplineChar *sc,int layer) {
 	xend += SVGMINLRPAD; // Give glyphs ending near or past the advance width some extra space
     fprintf(svg, "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" version=\"1.1\" viewBox=\"%d 0 %d %d\">\n",
 	    xstart, xend - xstart, (int) ceil(em_size));
-    fprintf(svg, "  <g transform=\"matrix(1 0 0 -1 0 %d)\">\n",
-	    sc->parent->ascent );
-    if ( sc->parent->multilayer || sc->parent->strokedfont || !svg_sc_any(sc,layer)) {
+    if ( ep->use_transform ) {
+	fprintf(svg, "  <g transform=\"matrix(1 0 0 -1 0 %d)\">\n", ascent );
+	scc = sc;
+    } else {
+	trans[3] = -1;
+	trans[5] = ascent;
+	if (sc->parent!=NULL) {
+	    scc = SplineCharCopy(sc, sc->parent, NULL);
+	    FVTrans(scc->parent->fv,scc,trans,NULL,
+	            fvt_nopreserve|fvt_dontmovewidth|fvt_noupdate);
+	} else {
+	    // This is probably an isolated layer, so just transform the splines
+	    // there and back
+	    scc = sc;
+	    MatInverse(invtrans,trans);
+	    SplinePointListTransformExtended(scc->layers[layer].splines, trans,
```

Comment:
So we modify in place then reverse that? Are you sure this is absolutely idempotent, and not subject to fp error/instability?

Reply:
FontForge does this in other places even without `tpmask_dontTrimValues` (an option I added). It's seen in `effects.c`, for example. I guess I could sub-in a copy if that was preferred.

Reply:
I added the copy